### PR TITLE
Add Pitch values for sound effects

### PR DESCRIPTION
**Pull request by windy1**
Example usage:

```java
world.playSound(SoundType.NOTE_PIANO, position, 1, Pitch.C1);
```

will play a middle C.

Signed-off-by: Walker Crouse <walkercrouse@hotmail.com>

### CODE REVIEW

**Comment by simon816** [comment]
There already is a [`NotePitches`](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/data/type/NotePitches.java) class

**Comment by windy1** [comment]
@simon816 Yes I understand, this is different though. The note block pitches are a value between 0 and 24 I believe while the sound effect pitches are between 0 and 2.

**Comment by simon816** [comment]
Perhaps there could just be a method `NotePitch#asDouble()` or something?
`world.playSound(SoundType.NOTE_PIANO, position, 1, NotePitches.C1.asDouble());`

**Comment by gabizou** [comment]
Note that by default, the pitch is actually separate from `NotePitch`es since the sounds are normally played at `1`, any higher or lower alters the pitch of the sound itself (in various ways, some sounds are played slower and some sounds are played faster).

**Comment by windy1** [comment]
I'll change it if you guys want I just think this way is more precise and clear. A developer is more likely to get confused if we combine these two things that are in reality two entirely different functions of the MC protocol.

**Comment by gabizou** [comment]
Perhaps you could rename `Pitch` to `PitchModulation` since that's realistically what it's doing (nothing to do with setting the sound's pitch, but more or less modifying the pitch).

**Comment by windy1** [comment]
@gabizou Commit updated

File: src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package org.spongepowered.api.effect.sound;
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java]: Missing license header

File: src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import com.flowpowered.math.vector.Vector3d;
+import org.spongepowered.api.effect.Viewer;
+
+/**
+ * Represents a pitch value for a {@link SoundType}. Minecraft's accepted pitches span two octaves from F#0 to F#2.
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java]: Maximum javadoc line length is 80

File: src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * @see Viewer#playSound(SoundType, Vector3d, double)
+ */
+public final class PitchModulation {
+    public static double FSHARP0 = 0;
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java]: Need newline after class declaration

File: src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    public static double AFLAT1 = GSHARP1;
+    public static double BFLAT1 = ASHARP1;
+    public static double CFLAT1 = B1;
+    public static double DFLAT2 = CSHARP2;
+    public static double EFLAT2 = DSHARP2;
+    public static double FFLAT2 = E2;
+    public static double GFLAT2 = FSHARP2;
+
+    private PitchModulation() {
+    }
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java]: Need newline after last member in the class. (before final `}`)

**Comment by windy1** [comment]
@simon816 updated

**Comment by Cybermaxke** [comment]
The fields should be final

**Comment by windy1** [comment]
whoops don't know how i missed that. fixed.

**Comment by windy1** [comment]
Do not pull this yet. The values are out of tune.

**Comment by windy1** [comment]
Okay fixed now.

File: src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.effect.sound;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.spongepowered.api.effect.Viewer;
+
+/**
+ * Represents a pitch value for a {@link SoundType}. Minecraft's accepted
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/effect/sound/PitchModulation.java]: probably should update the javadoc for what these do ;)

**Comment by windy1** [comment]
@gabizou updated

**Comment by gabizou** [comment]
Wait why was this closed? @windy1

**Comment by windy1** [comment]
I screwed up my master branch on my fork. I'll resubmit it when i get home later tonight.

**Pull request closed without merging**